internal/client/registry/apis/versions/artifacts.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package versions
+
+import (
+	"net/url"
+	"path"
+
+	"internal/apiclient"
+	"internal/client/registry/common"
+)
+
+type Action uint8
+
+const (
+	CREATE Action = iota
+	UPDATE
+)
+
+// CreateArtifact
+func CreateArtifact(apiName string, apiVersion string, artifactID string, name string, contents string,
+	labels map[string]string, annotations map[string]string,
+) (respBody []byte, err error) {
+	return createOrReplaceArtifact(apiName, apiVersion, artifactID, name, contents, labels, annotations, CREATE)
+}
+
+// ReaplceArtifact
+func ReplaceArtifact(apiName string, apiVersion string, artifactID string, name string, contents string,
+	labels map[string]string, annotations map[string]string,
+) (respBody []byte, err error) {
+	return createOrReplaceArtifact(apiName, apiVersion, artifactID, name, contents, labels, annotations, UPDATE)
+}
+
+// DeleteArtifact
+func DeleteArtifact(apiName string, apiVersion string, name string) (respBody []byte, err error) {
+	u, _ := url.Parse(apiclient.GetApigeeRegistryURL())
+	u.Path = path.Join(u.Path, "apis", apiName, "versions", apiVersion, "artifacts", name)
+	respBody, err = apiclient.HttpClient(u.String(), "", "DELETE")
+	return respBody, err
+}
+
+// GetArtifact
+func GetArtifact(apiName string, apiVersion string, name string) (respBody []byte, err error) {
+	u, _ := url.Parse(apiclient.GetApigeeRegistryURL())
+	u.Path = path.Join(u.Path, "apis", apiName, "versions", apiVersion, "artifacts", name)
+	respBody, err = apiclient.HttpClient(u.String())
+	return respBody, err
+}
+
+// GetArtifactContents
+func GetArtifactContents(apiName string, apiVersion string, name string) (respBody []byte, err error) {
+	u, _ := url.Parse(apiclient.GetApigeeRegistryURL())
+	u.Path = path.Join(u.Path, "apis", apiName, "versions", apiVersion, "artifacts", name+":getContents")
+	respBody, err = apiclient.HttpClient(u.String())
+	return respBody, err
+}
+
+// ListArtifacts
+func ListArtifacts(apiName string, apiVersion string, pageSize int,
+	pageToken string, filter string, orderBy string,
+) (respBody []byte, err error) {
+	u, _ := url.Parse(apiclient.GetApigeeRegistryURL())
+	u.Path = path.Join(u.Path, "apis", apiName, "versions", apiVersion, "artifacts")
+	return common.ListArtifacts(u, pageSize, pageToken, filter, orderBy)
```

Comment:
I see this error here:
```
gorun registry apis versions artifacts list --org=$ORG --api-name=ui-api-id-1
{
  "error": {
    "code": 400,
    "message": "Invalid JSON payload received. Unknown name \"pageSize\": Cannot bind query parameter. Field 'pageSize' could not be found in request message.",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.BadRequest",
        "fieldViolations": [
          {
            "description": "Invalid JSON payload received. Unknown name \"pageSize\": Cannot bind query parameter. Field 'pageSize' could not be found in request message."
          }
        ]
      }
    ]
  }
}
```

Reply:
The command is incorrect. I have added validations now. `api-name` and `api-version` are mandatory parameters.